paddle/cinn/hlir/dialect/operator/transforms/group_merge/substitute_dim_expr_based_on_constraint_pass.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2024 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "paddle/pir/include/pass/pass.h"
+
+namespace cinn {
+namespace dialect {
+namespace ir {
+
+// This is a helper pass for substituting the dim expr express based on the
+// constraints symbol::Equal<symbol::DimExpr>.
+std::unique_ptr<::pir::Pass> CreateSubstituteDimExprBasedOnConstraintPass();
```

Comment:
```suggestion
std::unique_ptr<::pir::Pass> CreateSubstituteDimExprBasedOnConstraintsPass();
```

`Constraint` -> `Constraints`，用复数更好些，其他地方也检查下

Reply:
已修改